### ISSUE TITLE
[Regression] Channel/user search results don't show real names of users

### ISSUE THREAD

**Issue by Gummikavalier** [RocketChat/Rocket.Chat#21849]
### Description:

Users that you have not direct messages with yet are not shown in the channel/user search listing with their real names.
The regression started in RC 3.14.0.

### Steps to reproduce:

1. Enable use of real names under `Admin -> Layout -> Use Real Name`
2. Seach for users in the channel search field
3. Look for users that you haven't started direct messaging yet

### Expected behavior:

All users are shown with real names.

### Actual behavior:

Users that you haven't chatted with yet are shown with usernames.
![Screenshot from 2021-04-28 18-58-15](https://user-images.githubusercontent.com/6958942/116435745-616a2e80-a854-11eb-88be-c641737c96b6.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.14.0
- Operating System: CentOS7
- Deployment Method: tar
- Number of Running Instances: 2
- DB Replicaset Oplog: Yes
- NodeJS Version: 12.22.1
- MongoDB Version: 4.0

### Client Setup Information

- Desktop App or Browser Version: All browsers

**Comment by Jeanstaquet** [comment]
Hello @Gummikavalier, do you have more infos ? On my side it works as it should.

**Comment by Gummikavalier** [comment]
Hi @Jeanstaquet . Not  much to add to it really. I have that specific test account `099fnd2ee` with real name `Test Account`. I created it as a normal local account to simplify the scenarios for other long standing similar issues (such as in reactions). It happens with Oauth provisioned accounts too.

As long as there does not already exist a direct discussion room between me and the account in the search result, the result always shows the username instead of real name (for any such account). If I select one, the username gets resolved as a real name, but only after the room has been established. So it is tricky to choose the correct person from a longer list of search results.

Using capitals in the search don't matter either. I also tried at open.rocket.chat but that is not very good at testing this, as they have a huge amount of people with really short usernames and real names, and both being the same.

**Comment by Gummikavalier** [comment]
This issue is still valid for all versions since 3.14.x. Open.rocket.chat is affected too, as depicted below.

![Screenshot from 2021-07-03 13-58-20](https://user-images.githubusercontent.com/6958942/124352114-0e6f7780-dc07-11eb-9e50-54e32364d0e3.png)

**Comment by debdutdeb** [comment]
Closing in favor of https://github.com/RocketChat/Rocket.Chat/issues/22602
